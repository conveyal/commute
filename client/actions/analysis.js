import moment from 'moment'
import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addAnalysis = createAction('add analysis')
export const createAnalysis = (newAnalysis, organizationId) => {
  newAnalysis.id = uuid.v4()
  newAnalysis.lastRunDateTime = moment().unix()
  const commuters = newAnalysis.commuters
  delete newAnalysis.commuters
  return [
    addAnalysis({ analysis: newAnalysis, organizationId }),
    push(`/organizations/${organizationId}/analysis/${newAnalysis.id}`),
    makeMockTrips(newAnalysis.id, organizationId, commuters)
  ] // TODO save to server
}

const deleteLocally = createAction('delete analysis')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/analysis/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteAnalysis = (id, organizationId) => [
  deleteLocally({ analysisId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]

const makeMockTrips = (analysisId, organizationId, commuters) => {
  const trips = []
  const modes = ['bike', 'car', 'transit', 'walk']
  const randRange = () => Math.random() * 0.4 + 0.8
  commuters.forEach((commuter) => {
    const curTrip = {
      commuterId: commuter.id
    }
    const tripMetricBase = Math.random()
    let bestValue = 9999999
    let bestMode
    modes.forEach((mode) => {
      // generate random trips
      const modePctMultiplier = randRange()
      const cost = tripMetricBase * 34.7 * modePctMultiplier * randRange()
      const distance = tripMetricBase * 69 * modePctMultiplier * randRange()
      const time = tripMetricBase * 7500 * modePctMultiplier * randRange()
      curTrip[mode] = {cost, distance, time}
      const tripValue = cost + time * 0.01
      if (tripValue < bestValue) bestMode = mode
    })
    curTrip.mostLikely = Object.assign({ mode: bestMode }, curTrip[bestMode])
    trips.push(curTrip)
  })
  return sendMockTrips({analysisId, organizationId, trips})
}
const sendMockTrips = createAction('receive mock calculated trips')
