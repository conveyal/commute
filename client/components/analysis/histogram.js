import React, {Component, PropTypes} from 'react'
import {Col, ControlLabel, FormControl, FormGroup, Grid, Panel, Row} from 'react-bootstrap'
import {
  Crosshair,
  DiscreteColorLegend,
  HorizontalGridLines,
  VerticalBarSeries,
  XAxis,
  YAxis
} from 'react-vis'

import ButtonLink from '../button-link'
import FlexiblePlot from '../flexible-plot'
import Icon from '../icon'
import analysisDefaults from '../../utils/analysisDefaults'
import {calcNumLessThan, getInitialSeries} from '../../utils/components'

const {metrics} = analysisDefaults
const metricsVals = Object.keys(metrics)

export default class Histogram extends Component {
  static propTypes = {
    // dispatch
    deleteAnalysis: PropTypes.func,

    // props
    analysis: PropTypes.object.isRequired,
    groupName: PropTypes.string.isRequired,
    site: PropTypes.object.isRequired
  }

  state = {
    crosshairValues: [],
    metric: 'time',
    series: getInitialSeries()
  }

  componentWillMount () {
    // calculate with selected metric
    this._calculateSeries({...this.state})
  }

  /**
   * Calculate series values given selected metric
   *
   * @param  {Object} newState The new state to work with
   */
  _calculateSeries (newState) {
    const arrayVals = this.props.analysis.tripVals
    const metricData = metrics[newState.metric]
    const multiplier = metricData.multiplier || 1

    // reset series data
    newState.series = getInitialSeries()

    // disallow transit distance metric
    if (newState.metric === 'distance') {
      newState.series = newState.series.filter((seriesMode) => seriesMode.mode !== 'transit')
    }

    newState.series = newState.series.map((seriesMode) => {
      seriesMode.data = []
      seriesMode.numInPreviousBins = 0
      return seriesMode
    })

    // iterate through all bins of histogram
    for (let i = 0; i < metricData.max; i += metricData.step) {
      const upper = i + metricData.step
      const curBinLabel = `${i} - ${upper}`

      // calculate number of trips by mode that fall inside current bin
      newState.series.forEach((seriesMode) => {
        // find total num to left
        const numLess = calcNumLessThan(arrayVals[seriesMode.mode][newState.metric],
          upper * multiplier)

        // add bin
        seriesMode.data.push({ x: curBinLabel, y: numLess - seriesMode.numInPreviousBins })

        // update for next iteration
        seriesMode.numInPreviousBins = numLess
      })
    }

    // add those off edge
    newState.series.forEach((seriesMode) => {
      // add bin with remnant
      seriesMode.data.push({
        x: `> ${metricData.max}`,
        y: arrayVals[seriesMode.mode][newState.metric].length - seriesMode.numInPreviousBins
      })
    })

    this.setState(newState)
  }

  /**
   * Format the title line of the crosshair.
   * @param {Array} values Array of values.
   * @returns {Object} The caption and the value of the title.
   * @private
   */
  _formatCrosshairTitle = (values) => {
    return {
      title: metrics[this.state.metric].unit,
      value: values[0].x
    }
  }

  /**
   * A callback to format the crosshair items.
   * @param {Object} values Array of values.
   * @returns {Array<Object>} Array of objects with titles and values.
   * @private
   */
  _formatCrosshairItems = (values) => {
    const {series} = this.state
    return values.map((v, i) => {
      return {
        title: series[i].title,
        value: v.y
      }
    })
  }

  /**
   * Handle the change of metric
   *
   * @param  {[type]} e [description]
   * @private
   */
  _handleMetricChange = (event) => {
    const newState = {...this.state}
    newState.metric = event.target.value
    this._calculateSeries(newState)
  }

  /**
   * Click handler for the legend.
   * @param {Object} item Clicked item of the legend.
   * @param {number} i Index of the legend.
   * @private
   */
  _legendClickHandler = (item, i) => {
    const {series} = this.state
    series[i].disabled = !series[i].disabled
    this.setState({series})
  }

  /**
   * Event handler for onMouseLeave.
   * @private
   */
  _mouseLeaveHandler = () => {
    this.setState({crosshairValues: []})
  }

  /**
   * Event handler for onNearestX.
   * @param {Object} value Selected value.
   * @param {number} index Index of the series.
   * @private
   */
  _nearestXHandler = (value, {index}) => {
    const {series} = this.state
    this.setState({
      crosshairValues: series.map(s => s.data[index])
    })
  }

  render () {
    const {_id: analysisId, name} = this.props.analysis
    const {groupName, site} = this.props
    const {crosshairValues, metric: selectedMetric, series} = this.state
    return (
      <Grid>
        <Row className='histogram-header'>
          <Col xs={12}>
            <h3>
              <span>{name}</span>
              <ButtonLink
                className='pull-right'
                to={`/analysis/${analysisId}`}
                >
                <Icon type='arrow-left' />
                <span>Back</span>
              </ButtonLink>
            </h3>
            <h3>Commute Metrics Histogram</h3>
          </Col>
          <Col xs={12} md={6}>
            <h4>
              <strong>Site:</strong>
              <span>{site.name}</span>
            </h4>
          </Col>
          <Col xs={12} md={6}>
            <h4>
              <strong>Group:</strong>
              <span>{groupName}</span>
            </h4>
          </Col>
        </Row>
        <Row>
          <Col xs={9}>
            <FlexiblePlot
              animation
              height={300}
              margin={{bottom: 70}}
              onMouseLeave={this._mouseLeaveHandler}
              xType='ordinal'
              >
              <Crosshair
                itemsFormat={this._formatCrosshairItems}
                titleFormat={this._formatCrosshairTitle}
                values={crosshairValues}
                />
              <HorizontalGridLines />
              {series
                .filter((s) => !s.disabled)
                .map((s, idx) =>
                  <VerticalBarSeries
                    color={s.color}
                    data={s.data}
                    key={`possibilities-series-${s.title}`}
                    onNearestX={this._nearestXHandler}
                    />
                )
              }
              <XAxis
                tickLabelAngle={-45}
                title={metrics[selectedMetric].unit}
                />
              <YAxis title='Number of Trips' />
            </FlexiblePlot>
          </Col>
          <Col xs={3}>
            <DiscreteColorLegend
              items={series}
              onItemClick={this._legendClickHandler}
              />
          </Col>
        </Row>
        <Row>
          <Col xs={12} md={8} className='histogram-settings'>
            <h4>Settings</h4>
            <Panel>
              <FormGroup controlId='histogram-metric-select'>
                <ControlLabel>Metric</ControlLabel>
                <FormControl
                  componentClass='select'
                  onChange={this._handleMetricChange}
                  value={selectedMetric}
                  >
                  {metricsVals.map((metric) => {
                    return <option value={metric}>{metrics[metric].label}</option>
                  })}
                </FormControl>
              </FormGroup>
            </Panel>
          </Col>
        </Row>
      </Grid>
    )
  }
}
