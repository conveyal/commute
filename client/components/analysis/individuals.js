import humanizeDuration from 'humanize-duration'
import {Browser} from 'leaflet'
import React, {Component, PropTypes} from 'react'
import {Button, Col, Grid, Row} from 'react-bootstrap'
import {BootstrapTable, TableHeaderColumn} from 'react-bootstrap-table'
import {Map as LeafletMap, TileLayer} from 'react-leaflet'
import {Link} from 'react-router'

import Icon from '../icon'
import {humanizeDistance} from '../../utils/components'
import {settings} from '../../utils/env'

export default class Individuals extends Component {
  static propTypes = {
    // dispatch
    deleteAnalysis: PropTypes.func,

    // props
    analysis: PropTypes.object.isRequired,
    commutersById: PropTypes.object.isRequired,
    groupName: PropTypes.string.isRequired,
    organizationId: PropTypes.string.isRequired,
    siteName: PropTypes.string.isRequired
  }

  constructor (props) {
    super(props)
    this.state = {
      position: settings.map.focus,
      zoom: 8
    }
  }

  _costRenderer = (cell, row) => {
    return `$${row.mostLikely.cost}`
  }

  _distanceRenderer = (cell, row) => {
    return humanizeDistance(row.mostLikely.distance)
  }

  _durationRenderer = (cell, row) => {
    return humanizeDuration(row.mostLikely.time * 60 * 1000)
  }

  _modeRenderer = (cell, row) => {
    return row.mostLikely.mode
  }

  _nameRenderer = (cell, row) => {
    return this.props.commutersById[row.commuterId].name
  }

  _onRowSelect = (row, isSelected) => {
    // TODO: show on map
    console.log(row)
    console.log('selected: ' + isSelected)
  }

  _ridematchesRenderer = (cell, row) => {
    return row.mostLikely.ridematches
  }

  render () {
    const {id, name} = this.props.analysis
    const {analysis, groupName, organizationId, siteName} = this.props
    const {position, zoom} = this.state
    return (
      <Grid>
        <Row className='individuals-header'>
          <Col xs={12}>
            <h3>
              <span>{name}</span>
              <Button className='pull-right'>
                <Link to={`/organizations/${organizationId}/analysis/${id}`}>
                  <Icon type='arrow-left' />
                  <span>Back</span>
                </Link>
              </Button>
            </h3>
            <h3>Individual Commutes</h3>
          </Col>
          <Col xs={12} md={6}>
            <h4>
              <strong>Site:</strong>
              <span>{siteName}</span>
            </h4>
          </Col>
          <Col xs={12} md={6}>
            <h4>
              <strong>Group:</strong>
              <span>{groupName}</span>
            </h4>
          </Col>
        </Row>
        <Row>
          <Col xs={8} style={{height: '400px'}}>
            <LeafletMap center={position} zoom={zoom}>
              <TileLayer
                url={Browser.retina &&
                  process.env.LEAFLET_RETINA_URL
                  ? process.env.LEAFLET_RETINA_URL
                  : process.env.LEAFLET_TILE_URL}
                attribution={process.env.LEAFLET_ATTRIBUTION}
                />
            </LeafletMap>
          </Col>
        </Row>
        <Row className='individuals-content'>
          <Col xs={12}>
            <h3>Individual Trips</h3>
            <BootstrapTable
              data={analysis.trips}
              selectRow={{
                mode: 'radio',
                clickToSelect: true,
                bgColor: 'rgb(238, 193, 213)',
                onSelect: this._onRowSelect
              }}
              >
              <TableHeaderColumn dataField='commuterId' isKey hidden />
              <TableHeaderColumn dataFormat={this._nameRenderer}>Name</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._modeRenderer}>Mode</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._durationRenderer}>Duration</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._distanceRenderer}>Distance</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._costRenderer}>Cost</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._ridematchesRenderer}>Ridematches</TableHeaderColumn>
            </BootstrapTable>
          </Col>
        </Row>
      </Grid>
    )
  }
}
