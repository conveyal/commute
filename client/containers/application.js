import {connect} from 'react-redux'
import {refreshUser} from 'mastarm/react/auth0'

import Application from '../components/application'

function mapStateToProps (state) {
  return {}
}

function mapDispatchToProps (dispatch) {
  return {
    refreshUserToken: () => refreshUser(dispatch)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Application)
