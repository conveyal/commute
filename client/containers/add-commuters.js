import {connect} from 'react-redux'

import {appendToGroup, createGroup} from '../actions/group'
import AddCommuters from '../components/add-commuters'

function mapStateToProps (state, props) {
  const {organization} = state
  const {params} = props
  const currentOrganizationId = params.organizationId
  const output = {
    organizationId: currentOrganizationId
  }
  if (params && params.groupId) {
    const currentOrganization = organization.organizationsById[currentOrganizationId]
    const currentGroup = currentOrganization.groupsById[params.groupId]
    Object.assign(output, {
      appendMode: true,
      organizationId: currentOrganizationId,
      group: currentGroup
    })
  }
  return output
}

function mapDispatchToProps (dispatch, props) {
  return {
    create: (group, organizationId) =>
      dispatch(createGroup(group, organizationId)),
    append: (commuters, groupId, organizationId) =>
      dispatch(appendToGroup(commuters, groupId, organizationId))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(AddCommuters)
