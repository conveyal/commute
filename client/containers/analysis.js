import {connect} from 'react-redux'

import {deleteAnalysis} from '../actions/analysis'
import _histogram from '../components/analysis/histogram'
import _individuals from '../components/analysis/individuals'
import _possibilities from '../components/analysis/possibilities'
import _summary from '../components/analysis/summary'

function mapStateToProps (state, props) {
  const {organization} = state
  const {params} = props
  const currentOrganizationId = params.organizationId
  const currentOrganization = organization.organizationsById[currentOrganizationId]
  const curAnalysis = currentOrganization.analysesById[params.analysisId]
  const group = currentOrganization.groupsById[curAnalysis.groupId]
  return {
    analysis: curAnalysis,
    commutersById: group.commutersById,
    groupName: group.name,
    organizationId: currentOrganizationId,
    siteName: currentOrganization.sitesById[curAnalysis.siteId].name
  }
}

function mapDispatchToProps (dispatch) {
  return {
    deleteAnalysis: (id, organizationId) => dispatch(deleteAnalysis(id, organizationId))
  }
}

export const Histogram = connect(mapStateToProps)(_histogram)
export const Individuals = connect(mapStateToProps)(_individuals)
export const Possibilities = connect(mapStateToProps)(_possibilities)
export const Summary = connect(mapStateToProps, mapDispatchToProps)(_summary)
