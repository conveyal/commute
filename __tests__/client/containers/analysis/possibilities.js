/* global describe, expect, it */

import {mount} from 'enzyme'
import {mountToJson} from 'enzyme-to-json'
import React from 'react'
import {Provider} from 'react-redux'

import {makeMockStore, mockStores} from '../../../test-utils/mock-data.js'

import {Possibilities} from '../../../../client/containers/analysis'

const mockStore = makeMockStore(mockStores.complexOrganization)

describe('Container > Analysis > Possibilities', () => {
  it('Analysis Possibilities View loads (base case)', () => {
    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <Possibilities
          params={{organizationId: '2', analysisId: '1'}}
          />
      </Provider>
    )
    expect(mountToJson(tree.find('.possibilities-header'))).toMatchSnapshot()
    expect(mountToJson(tree.find('.possibilities-legend-control'))).toMatchSnapshot()
    expect(mountToJson(tree.find('.table-bordered'))).toMatchSnapshot()
    expect(mountToJson(tree.find('.possibilities-settings'))).toMatchSnapshot()
  })

  /* it('Slide maximum travel time', () => {
    // Given a logged-in user is viewing the Possibilities Analysis View
    // When the user slide the maximum travel time slider
    // Then the bar chart should recalculate given the new constraints
  })

  it('Slide maximum distance', () => {
    // Given a logged-in user is viewing the Possibilities Analysis View
    // When the user slide the maximum distance slider
    // Then the bar chart should recalculate given the new constraints
  })

  it('Slide maximum cost', () => {
    // Given a logged-in user is viewing the Possibilities Analysis View
    // When the user slide the maximum cost slider
    // Then the bar chart should recalculate given the new constraints
  })

  it('Slide maximum carpool radius', () => {
    // Given a logged-in user is viewing the Possibilities Analysis View
    // When the user slide the maximum carpool radius slider
    // Then the bar chart should recalculate given the new constraints
  }) */
})
