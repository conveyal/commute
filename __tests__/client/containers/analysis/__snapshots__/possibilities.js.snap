exports[`Container > Analysis > Possibilities Analysis Possibilities View loads (base case) 1`] = `
<div
  className="possibilities-header row">
  <Col
    bsClass="col"
    componentClass="div"
    xs={12}>
    <div
      className="col-xs-12">
      <h3>
        <span>
          An Analysis
        </span>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="pull-right"
          disabled={false}>
          <button
            className="pull-right btn btn-default"
            disabled={false}
            type="button">
            <Link
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/organizations/2/analysis/1">
              <a
                onClick={[Function]}
                style={Object {}}>
                <Icon
                  type="arrow-left">
                  <i
                    className="fa fa-arrow-left fa-fw " />
                </Icon>
                <span>
                  Back
                </span>
              </a>
            </Link>
          </button>
        </Button>
      </h3>
      <h3>
        Possibilities Analysis
      </h3>
    </div>
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    md={6}
    xs={12}>
    <div
      className="col-md-6 col-xs-12">
      <h4>
        <strong>
          Site:
        </strong>
        <span>
          Acme Corp
        </span>
      </h4>
    </div>
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    md={6}
    xs={12}>
    <div
      className="col-md-6 col-xs-12">
      <h4>
        <strong>
          Group:
        </strong>
        <span>
          Mock Group
        </span>
      </h4>
    </div>
  </Col>
</div>
`;

exports[`Container > Analysis > Possibilities Analysis Possibilities View loads (base case) 2`] = `
<div
  className="possibilities-legend-control col-sm-3 col-xs-12">
  <DiscreteColorLegendItem
    colors={
      Array [
        "#12939A",
        "#79C7E3",
        "#1A3177",
        "#FF9833",
        "#EF5D28",
      ]
    }
    items={
      Array [
        Object {
          "color": "red",
          "data": Array [
            Object {
              "x": 1,
              "y": "0.0",
            },
          ],
          "disabled": false,
          "mode": "bike",
          "title": "Bike",
        },
        Object {
          "color": "brown",
          "data": Array [
            Object {
              "x": 1,
              "y": "100.0",
            },
          ],
          "disabled": false,
          "mode": "car",
          "title": "Car",
        },
        Object {
          "color": "orange",
          "data": Array [
            Object {
              "x": 1,
              "y": "0.0",
            },
          ],
          "disabled": false,
          "mode": "transit",
          "title": "Transit",
        },
        Object {
          "color": "blue",
          "data": Array [
            Object {
              "x": 1,
              "y": "0.0",
            },
          ],
          "disabled": false,
          "mode": "walk",
          "title": "Walk",
        },
      ]
    }
    onItemClick={[Function]}
    orientation="vertical">
    <div
      className="rv-discrete-color-legend vertical"
      style={
        Object {
          "height": undefined,
          "width": undefined,
        }
      }>
      <DiscreteColorLegendItem
        color="red"
        disabled={false}
        onClick={[Function]}
        orientation="vertical"
        title="Bike">
        <div
          className="rv-discrete-color-legend-item vertical clickable"
          onClick={[Function]}>
          <span
            className="rv-discrete-color-legend-item__color"
            style={
              Object {
                "background": "red",
              }
            } />
          <span
            className="rv-discrete-color-legend-item__title">
            Bike
          </span>
        </div>
      </DiscreteColorLegendItem>
      <DiscreteColorLegendItem
        color="brown"
        disabled={false}
        onClick={[Function]}
        orientation="vertical"
        title="Car">
        <div
          className="rv-discrete-color-legend-item vertical clickable"
          onClick={[Function]}>
          <span
            className="rv-discrete-color-legend-item__color"
            style={
              Object {
                "background": "brown",
              }
            } />
          <span
            className="rv-discrete-color-legend-item__title">
            Car
          </span>
        </div>
      </DiscreteColorLegendItem>
      <DiscreteColorLegendItem
        color="orange"
        disabled={false}
        onClick={[Function]}
        orientation="vertical"
        title="Transit">
        <div
          className="rv-discrete-color-legend-item vertical clickable"
          onClick={[Function]}>
          <span
            className="rv-discrete-color-legend-item__color"
            style={
              Object {
                "background": "orange",
              }
            } />
          <span
            className="rv-discrete-color-legend-item__title">
            Transit
          </span>
        </div>
      </DiscreteColorLegendItem>
      <DiscreteColorLegendItem
        color="blue"
        disabled={false}
        onClick={[Function]}
        orientation="vertical"
        title="Walk">
        <div
          className="rv-discrete-color-legend-item vertical clickable"
          onClick={[Function]}>
          <span
            className="rv-discrete-color-legend-item__color"
            style={
              Object {
                "background": "blue",
              }
            } />
          <span
            className="rv-discrete-color-legend-item__title">
            Walk
          </span>
        </div>
      </DiscreteColorLegendItem>
    </div>
  </DiscreteColorLegendItem>
  <Panel
    bsClass="panel"
    bsStyle="default"
    defaultExpanded={false}>
    <div
      className="panel panel-default"
      id={undefined}>
      <div
        className="panel-body">
        <FormGroup
          bsClass="form-group"
          controlId="histogram-metric-select">
          <div
            className="form-group">
            <ControlLabel
              bsClass="control-label"
              srOnly={false}>
              <label
                className="control-label"
                htmlFor="histogram-metric-select">
                Y Axis
              </label>
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              componentClass="select"
              onChange={[Function]}
              value="percent">
              <select
                className="form-control"
                id="histogram-metric-select"
                onChange={[Function]}
                type={undefined}
                value="percent">
                <option
                  value="percent">
                  Percent
                </option>
                <option
                  value="number">
                  Number of Commuters
                </option>
              </select>
            </FormControl>
          </div>
        </FormGroup>
      </div>
    </div>
  </Panel>
</div>
`;

exports[`Container > Analysis > Possibilities Analysis Possibilities View loads (base case) 3`] = `
<table
  className="table table-bordered">
  <tbody>
    <tr>
      <td>
        <span
          className="rv-discrete-color-legend-item__color"
          style={
            Object {
              "background": "red",
            }
          } />
        <span
          className="rv-discrete-color-legend-item__title">
          Bike - 0.0%
        </span>
      </td>
      <td>
        <span
          className="rv-discrete-color-legend-item__color"
          style={
            Object {
              "background": "brown",
            }
          } />
        <span
          className="rv-discrete-color-legend-item__title">
          Car - 100.0%
        </span>
      </td>
      <td>
        <span
          className="rv-discrete-color-legend-item__color"
          style={
            Object {
              "background": "orange",
            }
          } />
        <span
          className="rv-discrete-color-legend-item__title">
          Transit - 0.0%
        </span>
      </td>
      <td>
        <span
          className="rv-discrete-color-legend-item__color"
          style={
            Object {
              "background": "blue",
            }
          } />
        <span
          className="rv-discrete-color-legend-item__title">
          Walk - 0.0%
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Container > Analysis > Possibilities Analysis Possibilities View loads (base case) 4`] = `
<div
  className="possibilities-settings col-xs-12">
  <h4>
    Settings
  </h4>
  <Panel
    bsClass="panel"
    bsStyle="default"
    defaultExpanded={false}>
    <div
      className="panel panel-default"
      id={undefined}>
      <div
        className="panel-body">
        <p>
          Maximum Travel Time
        </p>
        <Slider
          allowCross={true}
          className=""
          defaultValue={30}
          disabled={false}
          dots={false}
          handle={
            <function CustomHandle(props) {
              var style = (0, _assign2.default)({ left: props.offset + '%' }, handleStyle);
              return (
                _react2.default.createElement('div', { style: style }, props.formatter(props.value)));
            
            }
              formatter={[Function]} />
          }
          included={true}
          marks={Object {}}
          max={200}
          min={0}
          onAfterChange={[Function]}
          onBeforeChange={[Function]}
          onChange={[Function]}
          prefixCls="rc-slider"
          pushable={false}
          range={false}
          step={1}
          tipFormatter={[Function]}
          tipTransitionName=""
          vertical={false}>
          <div
            className="rc-slider"
            onMouseDown={[Function]}
            onTouchStart={[Function]}>
            <div
              className="rc-slider-rail" />
            <Track
              className="rc-slider-track rc-slider-track-1"
              included={true}
              length={15}
              offset={0}
              vertical={false}>
              <div
                className="rc-slider-track rc-slider-track-1"
                style={
                  Object {
                    "left": "0%",
                    "visibility": "visible",
                    "width": "15%",
                  }
                } />
            </Track>
            <Steps
              dots={false}
              included={true}
              lowerBound={0}
              marks={Object {}}
              max={200}
              min={0}
              prefixCls="rc-slider"
              step={1}
              upperBound={30}
              vertical={false}>
              <div
                className="rc-slider-step" />
            </Steps>
            <CustomHandle
              className="rc-slider-handle rc-slider-handle-2 rc-slider-handle-upper"
              dragging={false}
              formatter={[Function]}
              noTip={false}
              offset={15}
              prefixCls="rc-slider"
              tipFormatter={[Function]}
              tipTransitionName=""
              tooltipPrefixCls={undefined}
              value={30}
              vertical={false}>
              <div
                style={
                  Object {
                    "background": "#fff",
                    "border": "2px solid #abe2fb",
                    "borderRadius": "3px",
                    "cursor": "pointer",
                    "fontSize": "14px",
                    "left": "15%",
                    "padding": "2px",
                    "position": "absolute",
                    "textAlign": "center",
                    "transform": "translate(-50%, -50%)",
                  }
                }>
                30 minutes
              </div>
            </CustomHandle>
            <Marks
              className="rc-slider-mark"
              included={true}
              lowerBound={0}
              marks={Object {}}
              max={200}
              min={0}
              upperBound={30}
              vertical={false}>
              <div
                className="rc-slider-mark" />
            </Marks>
          </div>
        </Slider>
      </div>
    </div>
  </Panel>
  <Panel
    bsClass="panel"
    bsStyle="default"
    defaultExpanded={false}>
    <div
      className="panel panel-default"
      id={undefined}>
      <div
        className="panel-body">
        <p>
          Maximum Distance
        </p>
        <Slider
          allowCross={true}
          className=""
          defaultValue={25}
          disabled={false}
          dots={false}
          handle={
            <function CustomHandle(props) {
              var style = (0, _assign2.default)({ left: props.offset + '%' }, handleStyle);
              return (
                _react2.default.createElement('div', { style: style }, props.formatter(props.value)));
            
            }
              formatter={[Function]} />
          }
          included={true}
          marks={Object {}}
          max={100}
          min={0}
          onAfterChange={[Function]}
          onBeforeChange={[Function]}
          onChange={[Function]}
          prefixCls="rc-slider"
          pushable={false}
          range={false}
          step={1}
          tipFormatter={[Function]}
          tipTransitionName=""
          vertical={false}>
          <div
            className="rc-slider"
            onMouseDown={[Function]}
            onTouchStart={[Function]}>
            <div
              className="rc-slider-rail" />
            <Track
              className="rc-slider-track rc-slider-track-1"
              included={true}
              length={25}
              offset={0}
              vertical={false}>
              <div
                className="rc-slider-track rc-slider-track-1"
                style={
                  Object {
                    "left": "0%",
                    "visibility": "visible",
                    "width": "25%",
                  }
                } />
            </Track>
            <Steps
              dots={false}
              included={true}
              lowerBound={0}
              marks={Object {}}
              max={100}
              min={0}
              prefixCls="rc-slider"
              step={1}
              upperBound={25}
              vertical={false}>
              <div
                className="rc-slider-step" />
            </Steps>
            <CustomHandle
              className="rc-slider-handle rc-slider-handle-2 rc-slider-handle-upper"
              dragging={false}
              formatter={[Function]}
              noTip={false}
              offset={25}
              prefixCls="rc-slider"
              tipFormatter={[Function]}
              tipTransitionName=""
              tooltipPrefixCls={undefined}
              value={25}
              vertical={false}>
              <div
                style={
                  Object {
                    "background": "#fff",
                    "border": "2px solid #abe2fb",
                    "borderRadius": "3px",
                    "cursor": "pointer",
                    "fontSize": "14px",
                    "left": "25%",
                    "padding": "2px",
                    "position": "absolute",
                    "textAlign": "center",
                    "transform": "translate(-50%, -50%)",
                  }
                }>
                25 miles
              </div>
            </CustomHandle>
            <Marks
              className="rc-slider-mark"
              included={true}
              lowerBound={0}
              marks={Object {}}
              max={100}
              min={0}
              upperBound={25}
              vertical={false}>
              <div
                className="rc-slider-mark" />
            </Marks>
          </div>
        </Slider>
      </div>
    </div>
  </Panel>
  <Panel
    bsClass="panel"
    bsStyle="default"
    defaultExpanded={false}>
    <div
      className="panel panel-default"
      id={undefined}>
      <div
        className="panel-body">
        <p>
          Maximum Cost
        </p>
        <Slider
          allowCross={true}
          className=""
          defaultValue={20}
          disabled={false}
          dots={false}
          handle={
            <function CustomHandle(props) {
              var style = (0, _assign2.default)({ left: props.offset + '%' }, handleStyle);
              return (
                _react2.default.createElement('div', { style: style }, props.formatter(props.value)));
            
            }
              formatter={[Function]} />
          }
          included={true}
          marks={Object {}}
          max={50}
          min={0}
          onAfterChange={[Function]}
          onBeforeChange={[Function]}
          onChange={[Function]}
          prefixCls="rc-slider"
          pushable={false}
          range={false}
          step={1}
          tipFormatter={[Function]}
          tipTransitionName=""
          vertical={false}>
          <div
            className="rc-slider"
            onMouseDown={[Function]}
            onTouchStart={[Function]}>
            <div
              className="rc-slider-rail" />
            <Track
              className="rc-slider-track rc-slider-track-1"
              included={true}
              length={40}
              offset={0}
              vertical={false}>
              <div
                className="rc-slider-track rc-slider-track-1"
                style={
                  Object {
                    "left": "0%",
                    "visibility": "visible",
                    "width": "40%",
                  }
                } />
            </Track>
            <Steps
              dots={false}
              included={true}
              lowerBound={0}
              marks={Object {}}
              max={50}
              min={0}
              prefixCls="rc-slider"
              step={1}
              upperBound={20}
              vertical={false}>
              <div
                className="rc-slider-step" />
            </Steps>
            <CustomHandle
              className="rc-slider-handle rc-slider-handle-2 rc-slider-handle-upper"
              dragging={false}
              formatter={[Function]}
              noTip={false}
              offset={40}
              prefixCls="rc-slider"
              tipFormatter={[Function]}
              tipTransitionName=""
              tooltipPrefixCls={undefined}
              value={20}
              vertical={false}>
              <div
                style={
                  Object {
                    "background": "#fff",
                    "border": "2px solid #abe2fb",
                    "borderRadius": "3px",
                    "cursor": "pointer",
                    "fontSize": "14px",
                    "left": "40%",
                    "padding": "2px",
                    "position": "absolute",
                    "textAlign": "center",
                    "transform": "translate(-50%, -50%)",
                  }
                }>
                $20
              </div>
            </CustomHandle>
            <Marks
              className="rc-slider-mark"
              included={true}
              lowerBound={0}
              marks={Object {}}
              max={50}
              min={0}
              upperBound={20}
              vertical={false}>
              <div
                className="rc-slider-mark" />
            </Marks>
          </div>
        </Slider>
      </div>
    </div>
  </Panel>
</div>
`;
